recipes/cylc-flow/meta.yaml

```diff
@@ -0,0 +1,70 @@
+{% set name = "cylc-flow" %}
+{% set version = "8.0a1" %}
+
+package:
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+
+source:
+  url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
+  sha256: "36f4f4363cb5cc7a3a613d2db6e84cc41c2825c6cd011191cc4821c5ca9c81a4"
+
+build:
+  number: 0
+  skip: True  # [not linux]
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv"
+
+requirements:
+  host:
+    - ansimarkup >=1.0.0
+    - colorama >=0.4
+    - graphene ==2.1.6
+    - jinja2 >=2.10.1
+    - markupsafe >=1.1
+    - metomi-isodatetime >=1!2.0.1
+    - pip
+    - protobuf >=3.7
+    - python >=3.7
+    - python-jose >=3.0
+    - pyzmq >=18.0
+  run:
+    - ansimarkup >=1.0.0
+    - colorama >=0.4
+    - cryptography
+    - graphene ==2.1.6
+    - jinja2 >=2.10.1
+    - markupsafe >=1.1
+    - metomi-isodatetime >=1!2.0.1
+    - protobuf >=3.7
+    - pyasn1
+    - python >=3.7
+    - python-jose >=3.0
+    - pyzmq >=18.0
+
+test:
+  imports:
+    - cylc.flow
+  requires:
```

Comment:
If you're not running tests with `pytest` this should be removed.

Reply:
I think the tests are being executed - at least in Travis - with `pytest`: https://github.com/cylc/cylc-flow/blob/9ff5b7719fe924a9a2c8c1b3446faaaf47a4c5b2/.travis.yml#L58

Even though we are still in the progress of migrating the old unit tests written with unittest to pytest.

So should we leave it?

Reply:
This section only applies when testing the package that has been built for conda in the conda-forge CI.

Another (possibly better) option is to add another line to the test commands to run `pytest`. If that's the case, you'll have to add [`source_files`](https://conda-forge.org/docs/maintainer/adding_pkgs.html?highlight=test#copying-test-files) to get the test code and possible a second entry in the sources section (as the tests aren't normally included in the tarball on PyPI):

```yaml
sources:
  - url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
    sha256: 36f4f4363cb5cc7a3a613d2db6e84cc41c2825c6cd011191cc4821c5ca9c81a4

  - url: https://github.com/cylc/cylc-flow/archive/{{ version }}.tar.gz
    sha256: ....
    folder: github_sources
```
the `source_files` are then `github_sources/tests`.

Reply:
Might be easier to leave the tests out for now then I guess. Most of our test coverage is done via functional tests - would be too hard to configure to run here I think (written in shell, long execution time, and some flaky tests).

Reply:
Unless it's a good practice to run the tests?

Reply:
Most Python packages only have an import test + checks for each command line program provided so this looks good enough to me. You can always add more if you decided they're needed to prevent packaging bugs.

Reply:
We might as well run the pytest battery, this should help flush out any obvious bugs, the functional tests may require configuration etc.

Reply:
I don't know if running all tests would work here... plus we are already using travis to run tests.

I'd vote to move it to a separate issue for discussion later.

Reply:
> I'd vote to move it to a separate issue for discussion later.

I've gone with this by removing the:
```yaml
requires:
  - pytest
```
specification in my commit.